avogadro/qtplugins/resetview/resetview.cpp

```diff
@@ -0,0 +1,225 @@
+#include <QtWidgets/QAction>
+#include <QtWidgets/QOpenGLWidget>
+
+#define CAMERA_NEAR_DISTANCE 13.35f // Experimental number
+
+namespace Avogadro {
+namespace QtPlugins {
+
+ResetView::ResetView(QObject* parent_)
+  : Avogadro::QtGui::ExtensionPlugin(parent_),
+    m_centerAction(new QAction(tr("Center"), this)),
+    m_viewToAxesAction(new QAction(tr("Align View to Axes"), this))
+{
+  connect(m_centerAction, SIGNAL(triggered()), SLOT(centerView()));
+  connect(m_viewToAxesAction, SIGNAL(triggered()), SLOT(alignToAxes()));
+}
+
+ResetView::~ResetView() {}
+
+QList<QAction*> ResetView::actions() const
+{
+  QList<QAction*> result;
+  return result << m_centerAction << m_viewToAxesAction;
+}
+
+QStringList ResetView::menuPath(QAction*) const
+{
+  return QStringList() << tr("&View");
+}
+
+void ResetView::setMolecule(QtGui::Molecule* mol)
+{
+  m_molecule = mol;
+}
+
+void ResetView::setCamera(Rendering::Camera* camera)
+{
+  m_camera = camera;
+}
+
+void ResetView::setActiveWidget(QWidget* widget)
+{
+  m_glWidget = widget;
+  connect(this, SIGNAL(updateRequested()), m_glWidget, SLOT(requestUpdate()));
+}
+
+bool ResetView::defaultChecks()
+{
+  if (m_molecule == nullptr || m_camera == nullptr)
+    return true;
+
+  // Check for 3D coordinates - it's useless to consider the camera otherwise
+  if (m_molecule->atomPositions3d().size() != m_molecule->atomCount())
+    return true;
+
+  // no need to animate when there are no atoms
+  if (m_molecule->atomCount() == 0) {
+    animationCameraDefault(false);
+    return true;
+  }
+  return false;
+}
+
+inline float getZDistance(const Eigen::Affine3f& projection, float x,
+                          Rendering::Projection perspective)
+{
+  if (perspective == Rendering::Projection::Perspective) {
+    float fov = 2.0f * std::atan(1.0f / projection(1, 1));
+    // float aspect = projection(1,1) / projection(0,0);
+    // tan (fov/2) = (x/2) / z -> z =  (x/2) / tan (fov/2)
+    return std::max(CAMERA_NEAR_DISTANCE, (x / 2.0f) / std::tan(fov / 2.0f));
+  } else {
+    return 3.0f * CAMERA_NEAR_DISTANCE;
+  }
+}
+inline void getBB(const Core::Array<Vector3>& mols, Vector3& min, Vector3& max)
+{
+  min = mols[0];
+  max = mols[0];
+  for (unsigned int i = 1; i < mols.size(); ++i) {
+    min.x() = std::min(mols[i].x(), min.x());
+    max.x() = std::max(mols[i].x(), max.x());
+
+    min.y() = std::min(mols[i].y(), min.y());
+    max.y() = std::max(mols[i].y(), max.y());
+
+    min.z() = std::min(mols[i].z(), min.z());
+    max.z() = std::max(mols[i].z(), max.z());
+  }
+}
+
+void ResetView::animationCameraDefault(bool animate)
+{
+  const Core::Array<Vector3> mols = m_molecule->atomPositions3d();
+  Vector3 min, max;
+
+  getBB(mols, min, max);
+
+  Eigen::Matrix3f linearGoal;
+  linearGoal.row(0) = Vector3f::UnitX();
+  linearGoal.row(1) = Vector3f::UnitY();
+  linearGoal.row(2) = Vector3f::UnitZ();
+  // calculate the translation matrix
+  Eigen::Affine3f* goal = new Eigen::Affine3f(linearGoal);
```

Comment:
Why `new`, where is this deleted?

Reply:
There is no need, true. This and the previous have no sense since I changed how the default camera is created